src/test/java/org/prebid/server/bidder/cpmstar/CPMStarBidderTest.java

```diff
@@ -0,0 +1,334 @@
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.banner;
+
+public class CPMStarBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://test.endpoint.com";
+
+    private CPMStarBidder cpmStarBidder;
+
+    @Before
+    public void setUp() {
+        cpmStarBidder = new CPMStarBidder(ENDPOINT_URL);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new CPMStarBidder("invalid_url"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorIfImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder
+                        .id("123")
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()))));
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = cpmStarBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Cannot deserialize instance");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldSkipInvalidImpressionAndAddError() {
+        // given
+        ExtPrebid<?, ExtImpCPMStar> ext = ExtPrebid.of(null, ExtImpCPMStar.of(12, 132));
+        Imp imp = givenImp(
+                impBuilder -> impBuilder
+                        .banner(null)
+                        .id("2")
+                        .ext(mapper.valueToTree(ext))
+                        .banner(Banner.builder().w(300).h(400).build())
+        );
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(
+                        imp,
+                        givenImp(impBuilder -> impBuilder
+                                .banner(null)
+                                .id("2")
+                                .ext(mapper.valueToTree(ext))
+                                .audio(Audio.builder().build()))))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = cpmStarBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput(
+                        "Only Banner and Video bid-types are supported at this time"));
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .flatExtracting(BidRequest::getImp)
+                .containsOnly(imp);
+    }
+
+    @Test
+    public void makeHttpRequestsShouldNotChangeBannerWidthAndHeightIfPresent() {
```

Comment:
This test is redundant. (We don't have any logic which concerns width and height in bidder)